spec/api-system-preferences-spec.js

```diff
@@ -35,6 +35,54 @@ describe('systemPreferences module', () => {
     })
   })
 
+  describe('systemPreferences.registerDefaults(defaults)', () => {
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
+    })
+
+    it('registers defaults', () => {
+      // to pass into registerDefaults
+      const defaultsDict = {
+        'one': 'ONE',
+        'two': 2,
+        'three': [1, 2, 3]
+      }
+
+      // to test type
+      const defaultsMap = [
+        { key: 'one', type: 'string', value: 'ONE' },
+        { key: 'two', value: 2, type: 'integer' },
+        { key: 'three', value: [1, 2, 3], type: 'array' }
+      ]
```

Comment:
Not a fan of this duplication of test data -- if you ever change the test, you have to make two changes instead of one.

Would be better to build one from the other, e.g. something like this (not tested):

```js
let defaultsDict = {}
defaultsMap.forEach(row => defaultsDict[row.key] = row.value);
```

Reply:
> let defaultsDict = {}

`const`

👍